src/sentry/tempest/tasks.py

```diff
@@ -0,0 +1,178 @@
+    for credentials in TempestCredentials.objects.all():
+        if credentials.latest_fetched_item_id is None:
+            fetch_latest_item_id.delay(credentials.id)
+        else:
+            poll_tempest_crashes.delay(credentials.id)
+
+
+@instrumented_task(
+    name="sentry.tempest.tasks.fetch_latest_item_id",
+    queue="tempest",
+)
+def fetch_latest_item_id(credentials_id: int) -> None:
+    # FIXME: Try catch this later
+    credentials = TempestCredentials.objects.select_related("project").get(id=credentials_id)
+    project_id = credentials.project.id
+    org_id = credentials.project.organization_id  # this should work?
+    client_id = credentials.client_id
+
+    try:
+        response_text = fetch_latest_id_from_tempest(
+            org_id=org_id,
+            project_id=project_id,
+            client_id=client_id,
+            client_secret=credentials.client_secret,
+        )
+
+        if response_text.isdigit():  # If response is a number the fetch was a success
+            credentials.latest_fetched_item_id = response_text
+            credentials.save(update_fields=["latest_fetched_item_id"])
+            return
+
+        # Things the user ought to know about
+        if response_text.startswith("Invalid credentials"):
+            credentials.message = "Seems like the provided credentials are invalid"
+            credentials.save(update_fields=["message"])
+            return
+        if response_text.startswith("IP address not allow-listed"):
+            credentials.message = "Seems like our IP is not allow-listed"
+            credentials.save(update_fields=["message"])
+            return
+
+        logger.info(
+            "Fetching the latest item id failed.",
+            extra={"org_id": org_id, "project_id": project_id, "client_id": client_id},
+        )
+    except Exception:
+        logger.info(
+            "Fetching the latest item id failed.",
+            extra={"org_id": org_id, "project_id": project_id, "client_id": client_id},
+        )
+
+
+@instrumented_task(
+    name="sentry.tempest.tasks.poll_tempest",
+    queue="tempest",
+)
+def poll_tempest_crashes(credentials_id: int) -> None:
+    # FIXME: Try catch this later
+    credentials = TempestCredentials.objects.select_related("project").get(id=credentials_id)
+    project_id = credentials.project.id
+    org_id = credentials.project.organization_id  # this should work?
+    client_id = credentials.client_id
+
+    try:
+        response_text = fetch_crashes_from_tempest(
+            org_id=org_id,
+            project_id=project_id,
+            client_id=client_id,
+            client_secret=credentials.client_secret,
+            dsn=ProjectKey.objects.get_or_create(
+                use_case=UseCase.TEMPEST, project=credentials.project
+            )[
+                0
+            ].get_dsn(),  # This does generate a dsn not sure if it will work in the grand scheme of things though.
```

Comment:
nit: for a better code readability, i would extract this into a variable outside of this function call, and then just pass dsn as the variable. It will make function call and all the args easier to read

Reply:
Also, how and when are we going to create this `ProjectKey` for Tempest related things?